route/access_rules_test.go

```diff
@@ -183,13 +183,13 @@ func TestAccessRules_AccessDeniedHTTP(t *testing.T) {
 			denied: true,
 		},
 		{
-			desc: "allowed xff and allowed remote addr",
+			desc: "single allowed xff and allowed remote addr",
```

Comment:
Only a single element of the XFF is allowed.  The others are denied.

Reply:
In the old code this was allowed as it only checked the first element.

Reply:
Ah, I thought that because the XFF address and the remote address are allowed the request should be allowed. But 1.2.3.4 isn't whitelisted. This wasn't obvious to me. I'll clean up the comments after the merge.

Reply:
Yes, that was part of the new behavior.  If any one of the addresses presented in the XFF fail to match an allow/whitelist rule the request is denied.  If any one of the addresses match a deny/blacklist rule the request is denied.